tests/test_empirical_tier2.py

```
@@ -0,0 +1,715 @@
+        if rank_of_primary == 1:
+            attack_successes += 1
+
+    attack_rate = attack_successes / N_SYNTH
+    random_baseline = 1.0 / N_DONORS
+    ratio = attack_rate / random_baseline
+    mean_primary_frac = float(np.mean(primary_fractions))
+
+    print(
+        f"\n[P2 region-mode] Attack success rate: {attack_rate:.3f} "
+        f"({attack_successes}/{N_SYNTH}), random baseline: {random_baseline:.4f}, "
+        f"ratio: {ratio:.1f}×"
```

Ruff (0.15.5)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)